merkledag/ipld.md

```diff
@@ -45,33 +53,83 @@ Suppose also that this path points to the object `{ "hello": "world" }`.
-O_1 = | "a": "QmV76pU..." |  whose hash value is QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k
-      +-------------------+
-                |
-                v
-      +-------------------+
-O_2 = | "b": "QmV76pU..." |  whose hash value is QmV76pUdAAukxEHt9Wp2xwyTpiCmzJCvjnMxyQBreaUeKT
-      +-------------------+
-                |
-                v
-      +-------------------+
-O_3 = | "c": "QmV76pU..." |  whose hash value is QmQmkZPNPoRkPd7wj2xUJe5v5DsY6MX33MFaGhZKB2pRSE
-      +-------------------+
-                |
-                v
-      +-------------------+
-O_4 = | "d": "QmV76pU..." |  whose hash value is QmWkyYNrN5wnHgX5vfs88q7QUaFKq52TVNTFeTzxm73UbT
-      +-------------------+
-                |
-                v
-      +-------------------+
-O_5 = | "hello": "world"  |  whose hash value is QmR8Bzg59Y4FGWHeu9iTYhwhiP8PHCNFiaGhP1UjywA43j
-      +-------------------+
+      +-----------------------------+
+O_1 = | "a": {"link": "QmV76pU..."} |  whose hash value is QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k
+      +-----------------------------+
+                     |
+                     v
+      +-----------------------------+
+O_2 = | "b": {"link": "QmQmkZP..."} |  whose hash value is QmV76pUdAAukxEHt9Wp2xwyTpiCmzJCvjnMxyQBreaUeKT
+      +-----------------------------+
+                     |
+                     v
+      +-----------------------------+
+O_3 = | "c": {"link": "QmWkyYN..."} |  whose hash value is QmQmkZPNPoRkPd7wj2xUJe5v5DsY6MX33MFaGhZKB2pRSE
+      +-----------------------------+
+                     |
+                     v
+      +-----------------------------+
+O_4 = | "d": {"link": "QmR8Bzg..."} |  whose hash value is QmWkyYNrN5wnHgX5vfs88q7QUaFKq52TVNTFeTzxm73UbT
+      +-----------------------------+
+                     |
+                     v
+           +-------------------+
+O_5 =      | "hello": "world"  |  whose hash value is QmR8Bzg59Y4FGWHeu9iTYhwhiP8PHCNFiaGhP1UjywA43j
+           +-------------------+
 ```
 
 This entire _merkle-path_ traversal is a unix-style path traversal over a _merkle-dag_ which uses _merkle-links_ with names.
 
+#### Accessing properties within IPLD objects
+
+Now, to travel within an IPLD object, we introduce a second separator: the dot (`.`). This separator can be used to avoid dereferencing `merkle-links` but travel within the IPLD object.
+
+For example, suppose we have this _merkle-path_:
+
+```
+/ipfs/QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k/a.b/c.d
+```
+
+The link will:
+
+- look up the first object `QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k` that we call `root`
+- Look up the key `root["a"]["b"]` and find here a _merkle-link_
+- Dereference this _merkle-link_ to get `object1`
+- Look up the key `object1["c"]["d"]` which will be returned as the result
+
+Note that if we added a trailing slash to the path (`/ipfs/QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k/a.b/c.d/`), we would perform a last _merkle-link_ dereferencing:
```

Comment:
yes, this is a strong solution possibility, for these paths. it is not without its problems, as a) people dislike this when applied to symlinks, and b) web servers and clients will often redirect trailing or non-trailing with varying rules.